cmd/remediation-aggregator/main.go

```diff
@@ -207,6 +207,9 @@ func createOrUpdateOneResult(crClient *complianceCrClient, owner metav1.Object,
 	}
 
 	res.SetLabels(labels)
+	if annotations != nil {
```

Comment:
I don't understand this check. Is it mean to prevent from setting nil annotations on updates?

Reply:
This function is shared. It creates both complianceCheckResults and Remediations. Remediations don't have annotations.

Reply:
Yes, I guess what I'm trying to ask is why is not OK to just call `SetLabels(nil)`

Reply:
I guess it would work. I just didn't wanna touch the annotations if there really isn't any. As the `SetAnnotations` function will create an empty map if it's `nil`: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go#L585

Reply:
OK, that makes sense. I ended up looking elsewhere (`https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go#L166`) where it was just a simple assignment.